pageObjects/careerPage.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+const { browser, protractor } = require("protractor");
+
+const ec = protractor.ExpectedConditions;
+
+
+
+class CareerPage {
+    constructor() {
+        this.logo = element(by.css('.header__logo'));
+        this.acceptButton = element(by.id('onetrust-accept-btn-handler'));
+        this.searchForm = element(by.id('new_form_job_search_1445745853_copy-keyword'));
+        this.locationArrow = element(by.css('.select2-selection__arrow'))
+        this.country = element(by.id('select2-new_form_job_search_1445745853_copy-location-result-kftm-all_Hungary'));
+        this.submitButton = element(by.css('.recruiting-search__submit'));
+        this.keywordInput = element(by.id('new_form_job_search_1445745853_copy-keyword'));
+        this.searchwordInput = element(by.css('.select2-search__field'));
+        this.selectresult = element(by.css('.select2-results__option select2-results__option--highlighted'));
+        this.applyButton = element(by.css('.search-result__item-apply:first-child'));
+        this.jobInfo = element(by.css('.search-result__item-name'));
+        this.contactButton = element(by.css('.button__content button__content--desktop'));
+        this.allLocations = element(by.css('span[title="All Cities in Hungary"]'));
+        this.url = 'https://www.epam.com/careers/job-listings?country=all'
+        this.allCountries = element.all(by.css('strong.select2-results__group'));
+        this.allCities = element.all(by.css('li.select2-results__option'));
+        this.suggestedCity = element(by.css('.select2-results__options.open'));
+                        
+    }
+
+    load() {
+        browser.get('https://www.epam.com/careers');
+        return browser.wait(ec.elementToBeClickable(this.logo), GLOBAL_TIMEOUT);
+    };
+
+    loadJobs() {
+
+        browser.get(this.url);
+        browser.wait(ec.elementToBeClickable(this.logo), GLOBAL_TIMEOUT);
+        return this.acceptAll();
+    }
+
+    acceptAll() {
+
+              
+        return browser.isElementPresent(this.acceptButton)
+            .then(isPresent => {
+                if (isPresent) {
+                    this.acceptButton.click();
+                }
+            });
+
+            
+    };
+
+    selectLocation() {
+        browser.wait(ec.elementToBeClickable(this.locationArrow), GLOBAL_TIMEOUT);
+        return this.locationArrow.click();
+       
+    }
+
+        
+  
+    clicksearchWord(){
+              browser.wait(ec.elementToBeClickable(this.searchwordInput), GLOBAL_TIMEOUT);
+        return this.searchwordInput.click();
+    }   
+
+    suggestedClick(){
+
+        browser.wait(ec.visibilityOf(this.suggestedCity), GLOBAL_TIMEOUT);
+        return this.suggestedCity.click();
+
+    }
+
+
+        
+    clickSubmit () {

```

Comment:
This step seems similar to clicksearchWord(). Do you think it could be merged into a general step, where the element to be clicked is passed as an argument? E.g. clickElement(element).